core/src/cluster_info.rs

```diff
@@ -1429,10 +1437,22 @@ impl ClusterInfo {
         bank_forks: Option<&Arc<RwLock<BankForks>>>,
         requests_receiver: &PacketReceiver,
         response_sender: &PacketSender,
+        thread_pool: &ThreadPool,
     ) -> Result<()> {
         //TODO cache connections
         let timeout = Duration::new(1, 0);
-        let reqs = requests_receiver.recv_timeout(timeout)?;
+        let mut requests = vec![requests_receiver.recv_timeout(timeout)?];
+        let mut num_requests = requests.last().unwrap().packets.len();
+        while let Ok(more_reqs) = requests_receiver.try_recv() {
```

Comment:
If this `try_recv` exits early, let's say at `.5` seconds, we reset to the `recv_timeout` and the MAX_GOSSIP_TRAFFIC counter resets, so we could potentially get a lot more than 1 GPBS if the requests are bursty and have enough of a gap to exit the `try_recv`.

Why don't we have a 1 second timer that we use to clear the `MAX_GOSSIP_TRAFFIC`. Check the timer after every `try_recv` and the `recv_timeout` and reset the `num_requests` counter

Reply:
As a follow up hopefully. This handles a worst case scenario. Strictly limiting it should help reduce wasted work as well.